src/encode.cpp

```diff
@@ -253,4 +255,46 @@ operator>>(MessageBuffer& buffer, PublishIntentEnd& msg)
   return true;
 }
 
+
+void
+operator<<(messages::MessageBuffer& msg, const Name& val)
+{
+  const auto& [hi, low] = HexEndec<128, 64, 64>::Decode(val);
+  msg << low;
+  msg << hi;
+}
```

Comment:
couple of observations
1. this still reveals the implementation of the name to have hi and low
2. Will it encode the value starting with 0x ?

Reply:
1. Still no, I just named these hi low to help myself, HexEndec knows nothing about Name's implementation
2. No, the return value of Decode is an array of uint64_t (unless specified as another unsigned int type)